src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/nomis/UserApiService.kt

```diff
@@ -73,14 +73,22 @@ class UserApiService(
     )
   }
 
-  fun findUserByUsername(username: String): PrisonUser? {
+  fun findUserByUsernameIgnoringErrors(username: String): PrisonUser? {
     if ("@" in username) {
       log.debug("Nomis not called with username as contained @: {}", username)
       return null
     }
     return serviceWebClientUtils.getIgnoreError("/users/${username.uppercase()}", PrisonUser::class.java)
   }
 
+  fun findUserByUsername(username: String): PrisonUser? {
+    if ("@" in username) {
+      log.debug("Nomis not called with username as contained @: {}", username)
```

Comment:
If we return null we don't fail gracefully.  I would've thought this should just throw a 404 exception if it contains @ to ensure the PrisonUser it always returned

Reply:
I will have a look thanks Karen.

Reply:
Its done thanks